libbeat/publisher/pipeline/pipeline.go

```diff
@@ -208,14 +209,29 @@ func (p *Pipeline) ConnectWith(cfg beat.ClientConfig) (beat.Client, error) {
 		return nil, err
 	}
 
+	reg := cfg.InputMetricsRegistry
+	if reg == nil {
+		// No registry, then create a registry which will not publish any metric,
+		// a 'discard' registry.
+		reg = monitoring.NewRegistry()
```

Comment:
Is there such a thing as a no-op registry? The concept of a 'discard' registry feels odd to me.

Reply:
it isn't a no-op, it's a registry that will not be published. I don't recall right now, but I don't think there is indeed a no-op registry. It's just for the code work regardless of having a valid registry or not. It's the same approach `inputmon` uses.